ai-integrations/langchain-graphrag.ipynb

```diff
@@ -0,0 +1,278 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "b5dcbf95-9a30-416d-afed-d5b2bf0e8651",
+   "metadata": {},
+   "source": [
+    "# GraphRAG with MongoDB and LangChain\n",
+    "\n",
+    "This notebook is a companion to the [GraphRAG with MongoDB and LangChain](https://www.mongodb.com/docs/atlas/atlas-vector-search/ai-integrations/langchain/graph-rag/) tutorial. Refer to the page for set-up instructions and detailed explanations.\n",
+    "\n",
+    "This notebook demonstrates a GraphRAG implementation using MongoDB Atlas and LangChain. Compared to vector-based RAG which structures your data as vector embeddings, GraphRAG structures data as a knowledge graph with entities and their relationships. This enables relationship-aware retrieval and multi-hop reasoning.\n",
+    "\n",
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/mongodb/docs-notebooks/blob/main/ai-integrations/langchain-graphrag.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "23f70093-83ea-4ecc-87db-2f2f89e546d7",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "pip install --quiet --upgrade pymongo langchain_community wikipedia langchain_openai langchain_mongodb pyvis"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d96955f9-a370-4f45-970d-ef187ee6195c",
+   "metadata": {},
+   "source": [
```

Comment:
Q: just curious why this page has a "Before you begin"; I don't see it in the others I checked

Reply:
The original scope for this repo did not include adding descriptions for each step, but ideally each notebook is a standalone resource, so these descriptions are nice to have. The contributor who helped create this notebook already included these descriptions, so might as well use what's already there!